File: demo-app/app/src/main/java/com/launchkey/android/whitelabel/demo/ui/fragment/CustomLogoutFragment2.java
package com.launchkey.android.whitelabel.demo.ui.fragment;

import android.app.ProgressDialog;
import android.os.Bundle;
import android.support.annotation.Nullable;

import com.launchkey.android.authenticator.sdk.SimpleOperationCallback;
import com.launchkey.android.authenticator.sdk.error.BaseError;
import com.launchkey.android.authenticator.sdk.session.SessionManager;
import com.launchkey.android.whitelabel.demo.R;
import com.launchkey.android.whitelabel.demo.util.Utils;

/**
 * Created by armando on 7/20/16.
 */
public class CustomLogoutFragment2 extends BaseDemoFragment {

    private ProgressDialog mLogoutDialog;

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        mLogoutDialog = new ProgressDialog(getActivity(), R.style.Theme_WhiteLabel_Dialog);
        mLogoutDialog.setIndeterminate(true);
        mLogoutDialog.setCancelable(false);
        mLogoutDialog.setMessage("Custom Logout...");

        logOut();
    }

    private void logOut() {

        Utils.show(mLogoutDialog);

        SessionManager
                .getInstance(getActivity())
                .endAllSessions(new SimpleOperationCallback() {

                    @Override
                    public void onResult(boolean successful, BaseError error, Object o) {

                        Utils.dismiss(mLogoutDialog);

                        if (successful) {
                            if (getActivity() != null && !getActivity().isFinishing()) {
                                getActivity().finish();
                            }
                        } else {
                            String message = Utils.getMessageForBaseError(error);
                            Utils.alert(getActivity(), "Error", message);
                        }
                    }
                });
    }
}

File: demo-app/app/src/kotlinApp/java/com/launchkey/android/authenticator/demo/ui/fragment/CustomLogoutFragment2.kt
package com.launchkey.android.authenticator.demo.ui.fragment

import android.app.ProgressDialog
import android.os.Bundle

import com.launchkey.android.authenticator.demo.R
import com.launchkey.android.authenticator.demo.util.Utils
import com.launchkey.android.authenticator.sdk.SimpleOperationCallback
import com.launchkey.android.authenticator.sdk.error.BaseError
import com.launchkey.android.authenticator.sdk.session.SessionManager

/**
 * Created by armando on 7/20/16.
 */
class CustomLogoutFragment2 : BaseDemoFragment() {

    private var mLogoutDialog: ProgressDialog? = null

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        mLogoutDialog = ProgressDialog(activity, R.style.Theme_WhiteLabel_Dialog)
        mLogoutDialog!!.isIndeterminate = true
        mLogoutDialog!!.setCancelable(false)
        mLogoutDialog!!.setMessage("Custom Logout...")

        logOut()
    }

    private fun logOut() {

        Utils.show(mLogoutDialog)

        SessionManager
                .getInstance(activity)
                .endAllSessions(object : SimpleOperationCallback<Any> {

                    override fun onResult(successful: Boolean, error: BaseError?, o: Any?) {

                        Utils.dismiss(mLogoutDialog)

                        if (successful) {
                            if (activity != null && !activity!!.isFinishing) {
                                activity!!.finish()
                            }
                        } else {
                            val message = Utils.getMessageForBaseError(error)
                            Utils.alert(activity, "Error", message)
                        }
                    }
                })
    }
}
